Assignment two

amitsindhwani opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
created logical models for 3 different prompts, created queries involving coalesce, window functions, did string manipulations, written queries involving Union, Cross Join, insert, update etc. 
## What did you learn from the changes you have made?
writing SQL queries pulling data from multiple table. almost every query had a different learning, not an easy assignment on SQL
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Yes tried multiple approaches, sometimes straight joins didn't work, had to work through subqueries. 
one of the subquery also took more than an hour to understand that I was returning multiple columns and update statement expects only value - amazing satisfaction after resolving that issue.
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
one of the subquery also took more than an hour to understand that I was returning multiple columns and update statement expects only value - amazing satisfaction after resolving that issue! hit and trial, solving one query at a time.

## How were these changes tested?
testing in SQLlite browser
## A reference to a related issue in your repository (if applicable)

## Checklist
- [ yes] I can confirm that my changes are working as intended

amitsindhwani committed:
assignment two involving logical model for 3 different prompts, queries involving coalesce function, windows functions, string manipulations, union, cross join, altering scheme, insert and update queries

amitsindhwani committed:
updated the sql and .md files

amitsindhwani committed:
added markdown

amitsindhwani commented:
took  6 mins late to submit, was facing issues with git to commit the .md file! finally resolved

efantinatti reviewed (changes requested):
ear Amit Kumar Sindhwani

Session 1:
1 - OK.


Session 2:
2 - OK.

Session 3:
3 - Partial.
-- DELETE - Q1: You must delete the same product name with an older insertion time. Be aware that -- INSERT must be the any of the products in the list.
-- UPDATE - Q1: Query looks good but COALESCE seems not taking effect. Can you figure out why?

Final:
4 - Please check these two items prior merging 'assignment-two' to 'main'.

Mark: 64 points.

amitsindhwani committed:
deleted the same product with an older insertion timestamp and applied coalesce to the subquery to ensure NULL replaced with 0

amitsindhwani commented:
Thanks, Ernani, for your feedback! I have updated the two queries based on your feedback! I also found the issue with COALESCE not working. This was not working for product ids without a match in the vendor_inventory table. So I applied at the subquery. Kindly review!

efantinatti reviewed (changes requested):
Hi Ait

I still see issues on DELETE and UPDATE.

The DELETE must delete the same product name with an older date.

The UPDATE is running into error. Result: near "SET": syntax error

amitsindhwani committed:
removed the previous delete queery

amitsindhwani commented:
Thanks, Ernani for reviewing my changes

On quick look, I checked again and updated delete query, update is working fine! Kindly review again.

For Delete query

For Delete, it shows that I forgot to remove the previous code so it’s not working correctly. I have removed the previous code and committed again!

Here is how I validated it, kindly check and let me know if I am missing something here. I inserted 3 records with product_id = 24.

After this I ran the delete query, that got executed successfully as shown in second column! And in the second row, when I executed the select query – it shows the last 2 records and the oldest one was deleted

Hope this is the correct output!

***@***.***
***@***.***
***@***.***




FOR Update query

There is no update to this query, I re-ran the query, it executed successfully and shows current-quantity column have zero values instead of NULL values!

***@***.***
***@***.***



Hope this time it works expected!

Regards
.***@***.***
www.rsystems.com<https://www.rsystems.com>
.
Amit Sindhwani
Dir.- Client Services & Technology

Mobile: +14168360142<tel:+14168360142>


From: Ernani Fantinatti ***@***.***>
Sent: December 23, 2024 4:22 PM
To: amitsindhwani/sql ***@***.***>
Cc: Amit Sindhwani ***@***.***>; Author ***@***.***>
Subject: Re: [amitsindhwani/sql] Assignment two (PR #2)

CAUTION: This message is sent from outside of "R SYSTEMS". Please DO NOT CLICK links or open attachments unless you recognize the source of this email and know the content is safe.

@efantinatti requested changes on this pull request.

Hi Ait

I still see issues on DELETE and UPDATE.

The DELETE must delete the same product name with an older date.

The UPDATE is running into error. Result: near "SET": syntax error

—
Reply to this email directly, view it on GitHub<https://protect.checkpoint.com/v2/r02/___https:/github.com/amitsindhwani/sql/pull/2%23pullrequestreview-2521154993___.YzJlOnJzeXN0ZW06YzpvOmU0ZGYwZjU2MGNjOWVmNzQ3N2I1YTI5YTNiYTdiYjczOjc6ZWFlYTozOWU1YTc5YjBlNTQwODcwMzI0NjI4N2Y1ZmQ1MTU2OTFkMzEyZTFiMzQyNmM1MDRmZDhkZTkzMzJkYWJhZThlOmg6VDpO>, or unsubscribe<https://protect.checkpoint.com/v2/r02/___https:/github.com/notifications/unsubscribe-auth/BM6JQW2BIUKJG6VWNUJKC332HB5GPAVCNFSM6AAAAABUCMYOMOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMRRGE2TIOJZGM___.YzJlOnJzeXN0ZW06YzpvOmU0ZGYwZjU2MGNjOWVmNzQ3N2I1YTI5YTNiYTdiYjczOjc6NDlhNTpjMTRmYTJlZWYzOTZhNmE4MjU5NDdkMTc4YTEwYTE1ODU3OGI5OGE2ZjFjNGY1MjM5ZTY5MGMwYTJjY2ZjNDE5Omg6VDpO>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

amitsindhwani commented:
Hello Ernani

I replied to the feedback email and the screenshots didn't came through, attaching the email for your kind review.

regards
Amit
[Delete and Update query screenshots.docx](https://github.com/user-attachments/files/18233700/Delete.and.Update.query.screenshots.docx)

efantinatti commented:
Hi Amitsindhwani

Please put the file in your repo.



On Mon, Dec 23, 2024 at 5:01 PM Amit Kumar Sindhwani <
***@***.***> wrote:

> Hello Ernani
>
> I replied to the feedback email and the screenshots didn't came through,
> attaching the email for your kind review.
>
> regards
> Amit
> Delete and Update query screenshots.docx
> <https://github.com/user-attachments/files/18233700/Delete.and.Update.query.screenshots.docx>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/amitsindhwani/sql/pull/2#issuecomment-2560360392>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABLMDBRUXOLLWGO6NUYP5YD2HCB5FAVCNFSM6AAAAABUCMYOMOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRQGM3DAMZZGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

amitsindhwani committed:
screenshots of queries executed

amitsindhwani commented:
Hello Ernani, Added the screenshots in repo folder - "02_activities/assignments/images/Delete and Update query screenshots.docx" Kindly review
